examples/todo-mvc-vue/src/todos.machine.ts

```diff
@@ -0,0 +1,127 @@
+import { createMachine, spawn, ActorRef } from 'xstate';
+// @ts-expect-error
+import uuid from 'uuid-v4';
+import { createTodoMachine } from './todoItem.machine';
+import { createModel } from 'xstate/lib/model';
+
+const createTodo = (title: string) => {
+  return {
+    id: uuid() as string,
+    title,
+    completed: false
+  };
+};
+
+interface Todo {
+  id: string;
+  title: string;
+  completed: boolean;
+  ref: ActorRef<any>;
+}
+
+const todosModel = createModel(
+  {
+    todo: '',
+    todos: [] as Todo[],
+    filter: 'all'
+  },
+  {
+    events: {
+      'NEWTODO.CHANGE': (value: string) => ({ value }),
+      'NEWTODO.COMMIT': (value: any) => ({ value }),
```

Comment:
Not sure why setting the value type a string gives an enormous TS error
```js
  NEWTODO.COMMIT': (value: string) => ({ value }), // this gives a TS error
```

Reply:
It's because of two things:

```diff
-          todos: (context, event: { value: string }) => {
+         todos: (context, event) => {
```

And this pesky conditional:

```ts
cond: (_, event) => event.value.trim().length
```

Gonna investigate this one.

Reply:
@davidkpiano got it!! With your indications I removed the manual string and changed the conditional to:
```diff
-  cond: (_, event) => event.value.trim().length
+  cond: (_, event) => !!event.value.trim().length

```
and now TypeScript is happy :)